Ensure receivexlog respects new configuration values after SIGHUP

nicois opened the pull request:
This modifies the receivexlog to use the updated configuration settings following a SIGHUP

# Why this way

When SIGHUP is issued, self.config is refreshed in all threads, but without this change, the new values are disregarded. This change ensures the updated config object is always used.

alanfranz-at-work reviewed (changes requested):
Some general comments:
* I think this new approach would require a lock to update the configuration; the current "do things in construction" is quite threadsafe, but re-reading the configuration at runtime may not be ok.
* We'd need a test.

nicois commented:
@alanfranz-at-work the tests are failing even when run against a "null" change: that is, my original intended commit, along with a revert commit. A later CI run, after I modified an assertion to more clearly show the failure reason, can be seen 
[here](https://github.com/aiven/pghoard/runs/6482816397?check_suite_focus=true#step:5:1077).

I'm hesitant of adding tests to an unstable test suite. I will modify this branch to consist of two commits, the second reverting the first, to better highlight the test instability.

alanfranz-at-work commented:
> @alanfranz-at-work the tests are failing even when run against a "null" change: that is, my original intended commit, along with a revert commit. It appears there is a problem with version pinning in the tests, leading to these failures. I'll see if there is a simple fix to resolve this problem.

Checking this.

> Eyeballing the code, the use of these value in this python script is thread-safe: these values are each used for stateless operations, primarily determining whether to stop or continue operation. I can add a couple of simple unit tests, one the existing tests are green again.

We cannot risk that a future change will fail and leave us with an inconsistent state. We should make sure that all config values are updated at the same time, and that they're not updated while being used.

Example:

```
        if now - self.last_disk_space_check < self.disk_space_check_interval:
            return

        bytes_free = self.get_disk_bytes_free()
        if not self.receiver_paused:
            if bytes_free < self.min_disk_space:
```

what happens if one section of the code runs with old version of disk_space_check_interval, and the other with an updated version of min_disk_space?

Possibly nothing, but the code is prone to failure.

nicois commented:
> what happens if one section of the code runs with old version of disk_space_check_interval, and the other with an updated version of min_disk_space?
> 
> Possibly nothing, but the code is prone to failure.

I've updated the code so new configuration changes are only assimilated at a specific point, meaning during each cycle the configuration values will remain consistent.

alanfranz-at-work commented:
@nicois please remember to re-request review after addressing comments, otherwise this can slip out of our view.

Please fix conflicts, then we'll make sure to have this reviewed.

nicois committed:
Allow SIGHUP to reload configuration settings

While SIGHUP was detected, the new configuration values were
not made available to subsystems which copied values from the
earlier configuration. Subsystems are now able to register
to receive configuration changes.

By using this approach, we also avoid potential race conditions
relating from configuration values changing at in appropriate times.
